crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -201,7 +203,8 @@ impl PhaseItem for Opaque3d {
 
     #[inline]
     fn sort_key(&self) -> Self::SortKey {
-        Reverse(FloatOrd(self.distance))
+        // Sort by pipeline, then by mesh to massively decrease drawcall counts in real scenes.
```

Comment:
Need to update this comment to mention bind groups. Also, was it not helpful to add bind group sorting for the prepasses?

Reply:
For the prepass and deferred it kept panicking on the unwrap when getting the material bindgroup.

We also may need to remove the unwrap for the opaque3d pass when getting the material bindgroup id, but I'm not sure if it makes sense that we'd be drawing things in the forward pass with no material bind group?

I'd like to remove the unwrap on getting the bind group id, but I'm not sure how the sorting would deal with a wrapped `Option<NonZeroU32>`.

Reply:
If you can't figure it out, I'd just add a TODO comment and we can do a followup PR